### PR TITLE
initial redesign: grow ffmprovsr

### PR DESCRIPTION
**Pull request by ablwr**
Related to #235, this puts the layout in a grid (and relational-sized instead of using pixels) instead of relying on Bootstrap, which will help in the future and sets a media size breakpoint so smaller screens see the new (thanks @kfrn!) table of contents at the top of a single column, but bigger screens push it to the sidebar on the left. Adds meta declaration for viewport width assessment.

I want to do more but I want to keep PRs small.

### CODE REVIEW

File: css/css.css [hunk 1]
```diff
@@ -1,9 +1,74 @@
...
+   "header"
+   "sidebar"
+   "content"
+   "footer"
+}
+
+.header {
+  grid-area: header;
+  text-align: center;
+
```
Comment (retokromer) [inline css/css.css]: blank line not needed

File: css/css.css [hunk 2]
```diff
@@ -1,9 +1,74 @@
+.grid {
+  display: grid;
+  align-content: center;
+  grid-template-areas:
+   "header"
+   "sidebar"
+   "content"
+   "footer"
```
Comment (retokromer) [inline css/css.css]: Nit: I guess (not verified), we have a `;` also at the end of the last item.

File: css/css.css [hunk 3]
```diff
@@ -1,9 +1,74 @@
...
+
+@media only screen and (min-width: 1px)  {
+.grid {
+  grid-template-columns: 1fr;
+  grid-template-areas:
+  "header"
+  "sidebar"
+  "content"
+  "footer";
+}
```
Comment (retokromer) [inline css/css.css]: I suggest to indent this level.

**Comment by ablwr** [comment]
I wouldn't get too caught up with perfecting this code since I anticipate it changing, what's more important is if the style/resizing is acceptable.

**Comment by ablwr** [comment]
I still made the changes though. 😅

File: index.html [hunk 4]
```diff
@@ -2,7 +2,7 @@
 <html lang="en">
   <head>
     <title>ffmprovisr</title>
-    <meta charset="utf-8">
+    <meta name="viewport" content="text/html, charset=UTF-8, width=device-width, initial-scale=1">
```
Comment (retokromer) [inline index.html]: Is this widely supported? I mean: can the `<meta charset="utf-8">` be grouped into this in HTML5?

Comment (ablwr) [reply index.html]: Yeah. Mostly this sets the default sizing so they can change as needed to fit onto different-sized devices. In this branch, only the h1 tag changes size based on how big the window is so it doesn't get smushed.

**Comment by retokromer** [comment]
I just saw it, going through the code…

File: index.html (hunk 4 above)
Comment (retokromer) [reply index.html]: Hmm… not sure. On Safari it gives now:

bildschirmfoto 2017-10-09 um 15 40 35

Comment (retokromer) [reply index.html]: It sets the standard charset of the browser, and does not understand to change. Putting again back that line, gives:

bildschirmfoto 2017-10-09 um 15 42 47

Comment (ablwr) [reply index.html]: Arg, thanks, hmm... I mean the charset=utf-8 *is* still in there. It might need an apostrophe instead of a comma. One sec!

**Review by retokromer — COMMENTED** [review]
Except the charset issue, it works fine on my side.

File: index.html (hunk 4 above)
Comment (ablwr) [reply index.html]: Sorry about that, I totally missed that charset="" was not scoped within the content=""!

**Review by retokromer — APPROVED** [review]
LGTM (tested only on my laptop)

**Comment by ablwr** [comment]
I will wait for @kfrn, who has a lot of CSS knowledge. Then I will work on tackling the bigger issue of replacing modals with inline...

**Comment by retokromer** [comment]
I suggest, once @kfrn has agreed, to merge this PR, and work on a new one.

**Comment by kfrn** [comment]
"A lot of CSS knowledge" lol. I shall have a look when I'm at home this evening.

**Comment by retokromer** [comment]
BTW: Considering the reaction of my 1st practical class yesterday (out of 8 this week), the changes already made go in the right direction. Thanks a lot!

**Comment by kfrn** [comment]
It looks great @ablwr ! Thanks so much for doing this!

**Review by kfrn — APPROVED** [review]

**Comment by kfrn** [comment]
Also - I'm really glad to hear that, @retokromer !

**Comment by ablwr** [comment]
Thanks so much to @kfrn!!!!!

**Comment by retokromer** [comment]
Thank you @ablwr and @kfrn for the clean-up! I just made a new release.

**Pull request merged**